File: frontend/src/app/core/services/shelf/models/shelf-details.dto.ts
import { ShelfType } from '../../../../shared/enums/shelf-type';

export interface ShelfDetailsDto {
  readonly id: number;
  readonly name: string;
  readonly type: ShelfType;
  readonly numberOfBooks: number;
}

export interface BookDto {
  readonly id: number;
  readonly title: string;
  readonly author: string;
}

File: backend/src/main/java/com/example/bookhub/dto/ShelfDetailsDto.java
package com.example.bookhub.dto;

import com.example.bookhub.enums.ShelfType;
import lombok.Data;

@Data
public class ShelfDetailsDto {

    private Long id;

    private String name;

    private ShelfType type;

    private int numberOfBooks;

    @Data
    public static class BookDto {

        private Long id;

        private String title;

        private String author;
    }
}
